libs/server/Lua/LuaRunner.cs

```diff
@@ -15,313 +18,854 @@ namespace Garnet.server
+            private readonly ScratchBufferManager bufferManager;
+            private byte* origin;
+            private byte* curHead;
+            private byte* curEnd;
+
+            internal RunnerAdapter(ScratchBufferManager bufferManager)
+            {
+                this.bufferManager = bufferManager;
+                this.bufferManager.Reset();
+
+                var scratchSpace = bufferManager.FullBuffer();
+
+                origin = curHead = (byte*)Unsafe.AsPointer(ref MemoryMarshal.GetReference(scratchSpace));
+                curEnd = curHead + scratchSpace.Length;
+            }
+
+#pragma warning disable CS9084 // Struct member returns 'this' or other instance members by reference
+            /// <inheritdoc />
+            public unsafe ref byte* BufferCur
+            => ref curHead;
+#pragma warning restore CS9084
+
+            /// <inheritdoc />
+            public unsafe byte* BufferEnd
+            => curEnd;
+
+            /// <summary>
+            /// Gets a span that covers the responses as written so far.
+            /// </summary>
+            public readonly ReadOnlySpan<byte> Response
+            {
+                get
+                {
+                    var len = (int)(curHead - origin);
+
+                    var full = bufferManager.FullBuffer();
+
+                    return full[..len];
+                }
+            }
+
+            /// <inheritdoc />
+            public void SendAndReset()
+            {
+                var len = (int)(curHead - origin);
+
+                // We don't actually send anywhere, we grow the backing array
+                bufferManager.GrowBuffer();
+
+                var scratchSpace = bufferManager.FullBuffer();
+
+                origin = (byte*)Unsafe.AsPointer(ref MemoryMarshal.GetReference(scratchSpace));
+                curEnd = origin + scratchSpace.Length;
+                curHead = origin + len;
+            }
+        }
+
+        const string LoaderBlock = @"
+import = function () end
+redis = {}
+function redis.call(...)
+    return garnet_call(...)
+end
+function redis.status_reply(text)
+    return text
+end
+function redis.error_reply(text)
+    return { err = 'ERR ' .. text }
+end
+KEYS = {}
+ARGV = {}
+sandbox_env = {
+    _G = _G;
+    _VERSION = _VERSION;
+
+    assert = assert;
+    collectgarbage = collectgarbage;
+    coroutine = coroutine;
+    error = error;
+    gcinfo = gcinfo;
+    -- explicitly not allowing getfenv
+    getmetatable = getmetatable;
+    ipairs = ipairs;
+    load = load;
+    loadstring = loadstring;
+    math = math;
+    next = next;
+    pairs = pairs;
+    pcall = pcall;
+    rawequal = rawequal;
+    rawget = rawget;
+    rawset = rawset;
+    redis = redis;
+    select = select;
+    -- explicitly not allowing setfenv
+    string = string;
+    setmetatable = setmetatable;
+    table = table;
+    tonumber = tonumber;
+    tostring = tostring;
+    type = type;
+    unpack = table.unpack;
+    xpcall = xpcall;
+
```

Comment:
This seems like a good set of built-ins. but I think we do need support for `struct` and `cmsgpack` -- these are commonly used in many popular Lua scripts such as BullMQ. See https://github.com/microsoft/garnet/issues/722#issuecomment-2412090364 for an initial sketch. Perhaps not necessary for this particular PR, though.

Reply:
Yeah, there's a broader "make sure everything exposed in Redis is available in Garnet"-pass to be done.  This PR syncs with Redis's allow-listed Lua builtins (since I was messing with the loader script anyway), but doesn't do anything for external libs or the various [`redis` object](https://redis.io/docs/latest/develop/interact/programmability/lua-api/#redis_object) extensions (except for `call`, `error_reply`,and `status_reply` which were already present).

I believe those should be separate PR, as they're mostly orthogonal to his work.